### ISSUE TITLE
'isinstance' check class cannot be 'self.__class__'

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#865]
**Describe the bug**
When attempting to check whether an object is an instance of a class where the class is extracted from the `__class__` attribute of the object, inside a method, Pyright reports that:

```
Second argument to "isinstance" must be a class or tuple of classes
  TypeVar or generic type with type arguments is not allowed
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `self.__class__` is not interpreted as a generic type or a type variable.

**Screenshots or Code**
```py
# pyright: strict

class Foo:
    def bar(self):
        return isinstance(object(), self.__class__)
```

**VS Code extension or command-line**
Extension, 1.1.54

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression I introduced in a recent change. It will be fixed in the next version of Pyright & Pylance.

**Comment by erictraut** [comment]
This is now implemented in Pyright 1.1.55, which I just published. It will also be included in the next release of Pylance.

**Comment by layday** [comment]
This appears to have been reintroduced in 1.1.79.

**Comment by erictraut** [comment]
@layday, you're correct. The recent regression was unrelated to the original cause, which is why our regression tests didn't catch it. I've fixed the issue and added this specific code to our regression tests so it doesn't happen again.

**Comment by layday** [comment]
Great, thank you!
